util/py/scripts/gh_testplan.py

```diff
@@ -0,0 +1,202 @@
+#!/usr/bin/env python3
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+import argparse
+import copy
+import json
+import logging
+import subprocess
+import sys
+import time
+
+import hjson
+
+UPSTREAM = "lowRISC/opentitan"
+COPYRIGHT_HEADER = """// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+"""
+
+flags = argparse.ArgumentParser(description="Testplan to github issue utility")
+flags.add_argument('--logging',
+                   default='info',
+                   choices=['debug', 'info', 'warning', 'error', 'critical'],
+                   help='Logging level')
+flags.add_argument('--gh_bin', default='gh', help="Github CLI binary")
+flags.add_argument('--repo', default=UPSTREAM, help="Upstream repo")
+flags.add_argument('--dry-run',
+                   action=argparse.BooleanOptionalAction,
+                   default=True,
+                   help='Do not perform any github API actions')
+flags.add_argument('testplan', type=str, help='The testplan to process')
+
+
+class Testplan(object):
+
+    def __init__(self, filename):
+        self.filename = filename
```

Comment:
Nit.
```suggestion
@dataclass
class Testplan(object):
    filename: str
```

Reply:
IMHO, dataclasses are not appropriate here.

The purpose of a dataclasss is to simply hold data and to not have any significant logic in the class.  The dataclass decorator helps with this purpose by adding all of the magical python "dunder" methods.

`Testplan`, however, represents the testplan file and performs some transformations on the data when loading/saving so that the in-memory representation is more convenient.

Similarly, the `GithubApi` class is a wrapper around interacting with github via the CLI tool (and, as discussed offline, if we're going to have a number of github-interacting tools in the codebase, we ought to switch to using `PyGithub` rather than wrapping the CLI).